docs/source/gs-processing/usage/emr.rst

```diff
@@ -0,0 +1,294 @@
+ready to run jobs:
+
+.. code-block:: bash
+
+    aws emr wait cluster-running --cluster-id j-XXXXXXXXXX --region ${REGION} && echo "Cluster ready"
+
+Log in to the leader and submit a GSProcessing job
+--------------------------------------------------
+
+To submit a job we can use a helper ``bash`` script, which we list below:
+
+.. code-block:: bash
+
+    # submit-gsp-job.sh
+    #!/usr/bin/env bash
+    set -euox pipefail
+
+    MY_BUCKET="enter-your-bucket-name-here"
+    REGION="bucket-region" # e.g. us-west-2
+    INPUT_PREFIX="s3://${MY_BUCKET}/gsprocessing-input"
+    NUM_EXECUTORS=2
+    OUTPUT_BUCKET=${MY_BUCKET}
+    GRAPH_NAME="small-graph"
+    CONFIG_FILE="gconstruct-config.json"
+    NUM_FILES="-1"
+    GSP_PATH="/path/to/graphstorm/graphstorm-processing"
+
+    ACCOUNT=$(aws sts get-caller-identity --query Account --output text)
+
+    REPOSITORY="graphstorm-processing-emr"
+    ARCH="x86_64"
+    TAG="latest-${ARCH}"
+    IMAGE="${ACCOUNT}.dkr.ecr.${REGION}.amazonaws.com/${REPOSITORY}:${TAG}"
+
+    S3_ENTRY_POINT="s3://${OUTPUT_BUCKET}/emr-scripts/distributed_executor.py"
+
+    # Upload the entry point, should be available locally
+    aws s3 cp ${GSP_PATH}/graphstorm_processing/distributed_executor.py ${S3_ENTRY_POINT}
+
+    export OUTPUT_PREFIX="s3://${OUTPUT_BUCKET}/gsprocessing/emr/${GRAPH_NAME}/${NUM_FILES}files/"
+
+    spark-submit --master yarn \
+        --deploy-mode cluster \
+        --conf spark.executorEnv.YARN_CONTAINER_RUNTIME_TYPE=docker \
+        --conf spark.executorEnv.YARN_CONTAINER_RUNTIME_DOCKER_IMAGE=${IMAGE} \
+        --conf spark.executorEnv.PYSPARK_PYTHON="/.pyenv/shims/python" \
+        --conf spark.yarn.appMasterEnv.YARN_CONTAINER_RUNTIME_TYPE=docker \
+        --conf spark.yarn.appMasterEnv.YARN_CONTAINER_RUNTIME_DOCKER_IMAGE=${IMAGE} \
+        --conf spark.yarn.appMasterEnv.PYSPARK_PYTHON="/.pyenv/shims/python" \
+        --num-executors ${NUM_EXECUTORS} \
+        ${S3_ENTRY_POINT} \
+            --config-filename ${CONFIG_FILENAME} \
+            --input-prefix ${INPUT_PREFIX} \
+            --output-prefix ${OUTPUT_PREFIX} \
+            --do-repartition True
+
+
+We will need to upload this helper script as well as the entry point
+script onto the Spark leader:
+
+.. code-block:: bash
+
+    aws emr put --cluster-id j-XXXXXXXXXX --key-pair-file /path/to/my-key-pair.pem \
+        --src submit-gsp-job.sh
+    aws emr put --cluster-id j-XXXXXXXXXX --key-pair-file /path/to/my-key-pair.pem \
+        --src /path/to/graphstorm/graphstorm-processing/graphstorm_processing/distributed_executor.py
+
+Once the cluster is launched we can use the key pair
+we created and the cluster ID to log into the Spark leader
+to submit jobs. We can do so by running:
+
+.. code-block:: bash
+
+    aws emr ssh --cluster-id j-XXXXXXXXXX --key-pair-file /path/to/my-key-pair.pem \
+        --region ${REGION}
+
+    bash submit-gsp-job.sh
+
+Ensure row counts are aligned and terminate cluster
```

Comment:
```suggestion
Ensure row counts are aligned and terminate the cluster
```

Reply:
Why we want to terminate the cluster?

Reply:
It's just a cleanup step, usually we include this in guides to remind users to not leave the cluster running, e.g. https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-gs.html#emr-getting-started-clean-up